### DISCUSSION TITLE
Error when installing a Sail project with Nova on a local machine

### DISCUSSION THREAD

**Discussion by JerryBels** [laravel/nova-issues#5728] [Q&A]
- Laravel Version: 10.15.0
- Nova Version: 4.26.4
- PHP Version: 8.2
- Database Driver & Version: MySQL 8.0
- Operating System and Version: Windows 11 (wsl - Ubuntu)
- Browser type and version: Chrome

### Description:
So, we have a Sail project to develop our app. I just added Nova without issues, and another dev cloned the repo and started the steps to install sail itself and start working on the project. Following the documentation : https://laravel.com/docs/10.x/sail#installing-composer-dependencies-for-existing-projects he ran the `docker run` command to install composer dependencies.
In the middle of the command, we get an error:
```
************************************************************************************
Unable to find Nova (Silver Surfer) license holder with the given email address: [].
************************************************************************************
    Failed to download laravel/nova from dist: The 'https://nova.laravel.com/dist/laravel/nova/laravel-nova-[whatever].zip' URL could not be accessed (HTTP 403): HTTP/2 403 
    Now trying to download from source
  - Syncing laravel/nova (4.26.5) into cache
```
Of course, there are no license informations available. And usually on those computers, Composer is not available in the local env - everything goes through containers. If we add manually an `auth.json` file with the relevant license information, everything works as intended. But I don't want to tell devs to do that every time they clone the project.

Is there a solution that would give devs ease of use? I don't think you can use `.env` variables in a json file, otherwise it would resolve the issue - git it without the license and use the information held in `.env`.

Thanks ahead.

**Comment by crynobone** [comment]
You should still have composer available on computers and setup global auth.json and add this: https://laracasts.com/discuss/channels/laravel/where-to-save-the-composer-github-credentials-when-using-sail?page=1&replyId=796261

**Reply by JerryBels** [reply]
Hi, and thanks. Seems very interesting! I'll be sure to dig into this. Although having to install composer on the machines, which means having to install PHP as well, may not be possible depending on the case. It also means having to send the license key to each and every developer when installing the project, even though they won't work on Nova at all or they will just tweak locally. It's a significant overhead when cloning the project, a hassle for big teams, and I feel like it's a bad move for the adoption of Nova.

**Reply by nathanheffley** [reply]
Just ran into this as well. Very frustrating to have to install Composer on my local machine in order to set up a dev environment that is 100% containerized otherwise.

**Reply by HWilliamRS** [reply]
Hi all,

I encountered the same issue yesterday and, after searching for a solution, I found a method worth documenting for others who might face the same problem.

To install Composer dependencies for existing applications using Laravel Sail, you can follow the instructions provided here: [Installing Composer Dependencies for Existing Applications](https://laravel.com/docs/11.x/sail#installing-composer-dependencies-for-existing-projects).

The basic command looks like this:

``` bash
docker run --rm \
    -u "$(id -u):$(id -g)" \
    -v "$(pwd):/var/www/html" \
    -w /var/www/html \
    laravelsail/php83-composer:latest \
    composer install --ignore-platform-reqs
```
However, if your project uses Laravel Nova, which requires a license, you can avoid manually typing the credentials by configuring Composer to store your Nova credentials. This can be done with the following command:

``` bash
composer config http-basic.nova.laravel.com your-nova-account-email@your-domain.com your-license-key
```
To combine both steps, you can use the following command:

``` bash
docker run --rm \
    -u "$(id -u):$(id -g)" \
    -v "$(pwd):/var/www/html" \
    -w /var/www/html \
    laravelsail/php83-composer:latest \
    /bin/sh -c "composer config http-basic.nova.laravel.com your-nova-account-email@your-domain.com your-license-key && composer install --ignore-platform-reqs"
```

This approach registers the Nova license before the Composer install begins, so you won't be prompted for credentials during the installation process.

I hope this helps!
